src/menu.tsx

```diff
@@ -148,35 +166,39 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
     });
 
     return (
-        <div
-            className={sprinkles({position: 'relative'})}
-            style={{
-                ...assignInlineVars({
-                    [styles.vars.width]: width ? `${width}px` : '100%',
-                    ...(itemsComputedProps
-                        ? {
-                              [styles.vars.top]: itemsComputedProps.top,
-                              [styles.vars.bottom]: itemsComputedProps.bottom,
-                              [styles.vars.right]: itemsComputedProps.right,
-                              [styles.vars.transformOrigin]: itemsComputedProps.transformOrigin,
-                              [styles.vars.maxHeight]: `${itemsComputedProps.maxHeight}px`,
-                          }
-                        : {}),
-                }),
-            }}
-            {...getPrefixedDataAttributes(dataAttributes, 'Menu')}
-        >
+        <div {...getPrefixedDataAttributes(dataAttributes, 'Menu')}>
+            {renderTarget({...targetProps, isMenuOpen})}
             {isMenuOpen ? (
-                <Overlay
-                    onPress={(e) => {
-                        cancelEvent(e);
-                        setIsMenuOpen(false);
-                    }}
-                    disableScroll
-                />
+                <Portal>
+                    <Overlay
+                        onPress={(e) => {
+                            cancelEvent(e);
+                            setIsMenuOpen(false);
+                        }}
+                        disableScroll
+                    >
+                        <div
+                            style={{
+                                ...assignInlineVars({
+                                    ...(itemsComputedProps
+                                        ? {
+                                              [styles.vars.top]: itemsComputedProps.top,
+                                              [styles.vars.bottom]: itemsComputedProps.bottom,
+                                              [styles.vars.left]: `${itemsComputedProps.left}px`,
+                                              [styles.vars.transformOrigin]:
+                                                  itemsComputedProps.transformOrigin,
+                                              [styles.vars.maxHeight]: `${itemsComputedProps.maxHeight}px`,
+                                              [styles.vars.width]: `${width}px`,
```

Comment:
some units with `px` and others as numbers

Reply:
Yes, and its intentional, I prefer to have a number when I want to restrict the type only to numbers, because I dont need "auto" or "100%".